pkg/lib/cockpit-components-simple-select.tsx

```diff
@@ -42,24 +51,38 @@ import {
   SelectList,
   SelectOption,
   SelectOptionProps,
-  SelectProps
+  SelectProps,
 } from '@patternfly/react-core/dist/esm/components/Select';
+import { Divider } from '@patternfly/react-core/dist/esm/components/Divider';
 import { MenuToggle, MenuToggleElement, MenuToggleProps } from '@patternfly/react-core/dist/esm/components/MenuToggle';
 
-export interface SimpleSelectOption extends Omit<SelectOptionProps, 'content'> {
+export interface SimpleSelectDividerOption {
+  decorator: "divider";
+
+  key: string | number;
+};
+
+export interface SimpleSelectMenuOption extends Omit<SelectOptionProps, 'content'> {
+  decorator?: undefined;
```

Comment:
OOI, why is that necessary? It means that the only possible value is `undefined`, right? How is that useful?

Reply:
This is to teach TypeScript how to distinguish between SimpleSelectMenuOption and SimpleSelectDividerOption: https://www.typescriptlang.org/docs/handbook/unions-and-intersections.html#discriminating-unions

Reply:
(I guess `decorator?: never` might be more ideomatic.)